File: test/clients/hmac.py
import hmac
import hashlib
import requests

# Shared secret key
SECRET_KEY = b'123'

# Request data
data = b"{\"hello\":\"world\"}"

# Generate HMAC
hmac_hash = hmac.new(SECRET_KEY, data, hashlib.sha256).hexdigest()

print(hmac_hash)

# Send request with HMAC in the headers
headers = {"Authorization": hmac_hash}
response = requests.get("http://localhost:8000/hmac", headers=headers)

# Print server response
print(response.status_code, response.text)

File: test/clients/hmac.cs
using System.Security.Cryptography;
using System.Text;

string ComputeHmacHex(string secretKey, string data)
{
    var keyBytes = Encoding.UTF8.GetBytes(secretKey);
    var dataBytes = Encoding.UTF8.GetBytes(data);

    using var hmac = new HMACSHA256(keyBytes);
    var hashBytes = hmac.ComputeHash(dataBytes);

    return Convert.ToHexString(hashBytes).ToLowerInvariant();
}

string secret = "123";
string data = "{\"hello\":\"world\"}";

string hmacHash = ComputeHmacHex(secret, data);
Console.WriteLine(hmacHash);